src/lib.rs

```diff
@@ -180,15 +180,55 @@ pub trait Specifier {
     ///
     /// This is the type that is used internally for computations.
     type Base: Default + private::PushBits + private::PopBits;
+
+    /// The type returned by getter functions.
+    ///
+    /// # Note
+    ///
+    /// Although this is usually the same as the Face type, in the case of
+    /// non-power-of-two enums (or other fields that may fail), this type is
+    /// a Result type.
+    type GetterReturn;
+
     /// The interface type of the specifier.
     ///
     /// # Note
     ///
     /// This is the type that is used for the getters and setters.
-    type Face: private::FromBits<Self::Base> + private::IntoBits<Self::Base>;
+    type Face: private::FromBits<Self::Base, Self::GetterReturn>
+        + private::IntoBits<Self::Base>;
 }
 
 /// The default set of predefined specifiers.
 pub mod specifiers {
     ::modular_bitfield_impl::define_specifiers!();
 }
+
+/// Error type used for working with enums
+#[derive(Debug, PartialEq)]
+pub enum EnumError {
+    /// Error returned when getting a field from a non-power-of-two enum when
+    /// the field does not match any of the variants.
+    UnknownValue(u32),
+}
+
+impl EnumError {
+    /// Returns whether this error has an attached raw value that can be
+    /// retrieved with raw_value.
+    pub fn has_raw_value(&self) -> bool {
+        match self {
+            Self::UnknownValue(_) => true,
+        }
+    }
+
+    /// If the error has an associated raw value, returns that raw value.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the error has no associated raw value.
+    pub fn raw_value(&self) -> u32 {
+        match self {
+            Self::UnknownValue(x) => *x,
```

Comment:
```suggestion
            Self::UnknownValue(&raw_value) => raw_value,
```

Reply:
Hm, I get this error when I match a reference:
```
   --> /home/lane/Downloads/modular-bitfield/src/lib.rs:231:32
    |
230 |         match self {
    |               ---- this expression has type `&EnumError`
231 |             Self::UnknownValue(&raw_value) => raw_value,
    |                                ^^^^^^^^^^
    |                                |
    |                                expected `u32`, found reference
    |                                help: you can probably remove the explicit borrow: `raw_value`
    |
    = note:   expected type `u32`
            found reference `&_`
```

Reply:
Ah yeah it probably needs to be `&Self::UnknownValue(..)` but I would consider this "ugly" so simply do:
`Self::UnknownValue(raw_value) => *raw_value,`